### PR TITLE
Add samba hook copyfile functionality

### PR DESCRIPTION
**Pull request by venufix**
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/samba/hooks/samba.py [hunk 1]
```diff
@@ -135,6 +135,9 @@ def open_file(
             file_type=file_type,
             **self._conn_kwargs,
         )
+    @wraps(smbclient.copyfile)
+    def copyfile(self, src, dst):
+        return smbclient.copyfile(self.__join_path(src), self.__join_path(dst), **self._conn_kwargs)
```
Comment (Taragolis) [inline airflow/providers/samba/hooks/samba.py:140]:
```suggestion
        return smbclient.copyfile(self._join_path(src), self._join_path(dst), **self._conn_kwargs)
```

**Review by Taragolis — CHANGES_REQUESTED** [review]
- Static checks failed, please follow this [Guide](https://github.com/apache/airflow/blob/main/STATIC_CODE_CHECKS.rst)
- Tests need to be fixed

**Comment by Taragolis** [comment]
@venufix, do you have a plan to continue work on this PR?

**Comment by venufix** [comment]
Unfortunately no, found another solution you fan actually get the samba client args from the hook and run the function

**Comment by Taragolis** [comment]
> Unfortunately no, found another solution you fan actually get the samba client args from the hook and run the function

No problem at all.

**Pull request closed without merging**